src/nemos/basis/_basis.py

```diff
@@ -0,0 +1,1406 @@
+        Y :
+            The basis function evaluated at the samples,
+            shape ``(n_samples[0], ... , n_samples[n], number of basis)``.
+
+        Raises
+        ------
+        ValueError
+            If the time point number is inconsistent between inputs or if the number of inputs doesn't match what
+            the Basis object requires.
+        ValueError
+            If one of the n_samples is <= 0.
+
+        Notes
+        -----
+        Setting ``indexing = 'ij'`` returns a meshgrid with matrix indexing. In the N-D case with inputs of size
+        :math:`M_1,...,M_N`, outputs are of shape :math:`(M_1, M_2, M_3, ....,M_N)`.
+        This differs from the numpy.meshgrid default, which uses Cartesian indexing.
+        For the same input, Cartesian indexing would return an output of shape :math:`(M_2, M_1, M_3, ....,M_N)`.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> import nemos as nmo
+
+        >>> # define two basis objects and add them
+        >>> basis_1 = nmo.basis.BSplineEval(10)
+        >>> basis_2 = nmo.basis.RaisedCosineLinearEval(15)
+        >>> additive_basis = basis_1 + basis_2
+
+        >>> # evaluate on a grid of 10 x 10 equi-spaced samples
+        >>> X, Y, Z = additive_basis.evaluate_on_grid(10, 10)
+
+        """
+        return super().evaluate_on_grid(*n_samples)
+
+
+class MultiplicativeBasis(Basis):
+    """
+    Class representing the multiplication (external product) of two Basis objects.
+
+    Parameters
+    ----------
+    basis1 :
+        First basis object to multiply.
+    basis2 :
+        Second basis object to multiply.
+
+    Attributes
+    ----------
+    n_basis_funcs :
+        Number of basis functions.
+
+    Examples
+    --------
+    >>> # Generate sample data
+    >>> import numpy as np
+    >>> import nemos as nmo
+    >>> X = np.random.normal(size=(30, 3))
+
+    >>> # define two basis and multiply
+    >>> basis_1 = nmo.basis.BSplineEval(10)
+    >>> basis_2 = nmo.basis.RaisedCosineLinearEval(15)
+    >>> multiplicative_basis = basis_1 * basis_2
+
+    >>> # Can multiply or add another basis to the AdditiveBasis object
+    >>> # This will cause the number of output features of the result basis to grow accordingly
+    >>> basis_3 = nmo.basis.RaisedCosineLogEval(100)
+    >>> multiplicative_basis_2 = multiplicative_basis * basis_3
+    """
+
+    def __init__(self, basis1: Basis, basis2: Basis) -> None:
+        self.n_basis_funcs = basis1.n_basis_funcs * basis2.n_basis_funcs
+        super().__init__(self.n_basis_funcs, mode="eval")
+        self._n_input_dimensionality = (
+            basis1._n_input_dimensionality + basis2._n_input_dimensionality
+        )
+        self._n_basis_input = None
+        self._n_output_features = None
+        self._label = "(" + basis1.label + " * " + basis2.label + ")"
+        self._basis1 = basis1
+        self._basis2 = basis2
+        BasisTransformerMixin.__init__(self)
```

Comment:
this line should be removed right? BasisTransformerMixin doesn't have an init method (and this is the only place it's called, which is weird, because it's inherited explicitly in Basis)